src/models/melodyDetector.coffee

```diff
@@ -16,37 +16,18 @@ module.exports =
 
 class MelodyDetector
   constructor: (@settings) ->
-    @recognizer = new AubioPitch(@settings.filePath)
+    @recognizer = new AubioPitch(@settings.filePath, @settings.options)
 
   ###
-  Generate the melody from the output of the recognizer.
-  returns a promise that resolves in something like [
-   { timestamp: 0, note: "r" },
-   { timestamp: 1.05, note: "c#4" }
-  ]
+  Generates the melody using the output of the recognizer
+  and a dynamic list of chained postprocessors (_.flow).
   ###
   getMelody: =>
-    @recognizer.execute().then (output) =>
-      notes = @_removeRepeatedNotes output.map(@_detectNote)
-
-      notesWithDuration = @_addDurationToNotes notes
-
-      new Melody(@settings.tempo, notesWithDuration)
-
-  _detectNote: (sampledNote) =>
-    detectedNote = noteDictionary.whatIs sampledNote.frequency
-    _.assign sampledNote,
-      name: detectedNote?.note
-
-  _removeRepeatedNotes: (notes) =>
-    notes.reject (sampledNote, i) =>
-      if not sampledNote.name? then return true
+    @recognizer.execute().then (samples) =>
+      postProcessors = readDir("#{__dirname}/postprocessors")
+        .reject (file) => file.startsWith(".") or _.contains(file, "spec")
+        .map (file) => require("./postprocessors/#{file}").bind @, @settings
 
-      detectedNote = (it) => it?.name
-      detectedNote(sampledNote) is detectedNote(notes[i - 1])
+      notes = (_.flow.apply @, postProcessors)(samples)
 
```

Comment:
1)
"bind" es un método que tienen las funciones para poder hacer *aplicación parcial* (el que vimos en paradigmas). Es decir, crear una nueva función igual pero con ese argumento ya asociado ("bindeado") al parámetro. Por ejemplo: function suma(a, b)  { return a + b; } ; sumarTres = suma.bind(null, 3) me devuelve una función que suma 3 con el número que le pases: sumarTres(2) devuelve 5.
El primer parámetro que pide es la referencia a this que va a tener la función, que como los postprocessors no usan this le paso cualquiera (el this actual o sea @).
Con esto les bindeo las settings a todos los postprocessors, que es su primer argumento. Y como resultado el map se obtiene una lista de funciones que solo esperan los samples. 

2) Leete en lodash qué hace flow. Genera una nueva función que va a invocar a la lista de funciones que le pases, pasandole a cada una como argumento lo que devolvió la anterior. Luego de generar esa función la invoco (segundo paréntesis) pasándole el argumento de la primera, que son los samples. 

3) Dalee. Lo muevo a un método (getPostprocessors) y lo guardo en el constructor así queda más claro y permite que a futuro se pueda testear.